### PR TITLE
[mfx_dispatch_test] Plugin test cases extension

### PR DESCRIPTION
**Pull request by semelianova**

### CODE REVIEW

File: tests/mfx_dispatch/linux/mfx_dispatch_test_cases_plugins.cpp [hunk 1]
```diff
@@ -80,6 +95,140 @@ TEST_P(DispatcherPluginsTestParametrized, ShouldLoadFromGoodPluginsCfgCorrectly)
     ASSERT_EQ(sts, MFX_ERR_NONE);
 }
 
-INSTANTIATE_TEST_CASE_P(EnumeratingPlugins, DispatcherPluginsTestParametrized, ::testing::Range((size_t) 0, TEST_PLUGIN_COUNT));
+INSTANTIATE_TEST_CASE_P(EnumeratingPlugins, DispatcherPluginsTestParametrized, ::testing::Range((size_t)0, TEST_PLUGIN_COUNT));
+
+TEST_F(DispatcherPluginsTestParametrized, ShouldFailIfPluginHasNoSymbols)
```
Comment (vshampor) [inline tests/mfx_dispatch/linux/mfx_dispatch_test_cases_plugins.cpp]: The test fixture used here and in other similar spots should be `DispatcherPluginsTest`, not `DispatcherPluginsTestParametrized`.

Comment (semelianova) [reply tests/mfx_dispatch/linux/mfx_dispatch_test_cases_plugins.cpp]: I agree with you. I will fix it.

File: tests/mfx_dispatch/linux/mfx_dispatch_test_cases_plugins.cpp [hunk 2]
```diff
@@ -41,6 +41,22 @@ TEST_F(DispatcherPluginsTest, ShouldSearchForPluginsCfgByCorrectPath)
     MFXVideoUSER_Load(session, &uid, 0);
 }
 
+TEST_F(DispatcherPluginsTest, ShouldFailIfPluginIsNotFound)
+{
+    MockCallObj& mock = *g_call_obj_ptr;
+
+    std::string path = std::string(MFX_PLUGINS_CONF_DIR) + "/plugins.cfg";
```
Comment (mgonchar) [inline tests/mfx_dispatch/linux/mfx_dispatch_test_cases_plugins.cpp]: too many strings constructors actually:
```
std::string path = // move constructor
std::string(MFX_PLUGINS_CONF_DIR) // constructor
 +  // constructor called by operator +
"/plugins.cfg"; // constructor
```
I think following should work:
```
std::string path(MFX_PLUGINS_CONF_DIR "/plugins.cfg");
```

File: tests/mfx_dispatch/linux/mfx_dispatch_test_cases_plugins.cpp [hunk 3]
```diff
@@ -41,6 +41,22 @@ TEST_F(DispatcherPluginsTest, ShouldSearchForPluginsCfgByCorrectPath)
...
+    EXPECT_CALL(mock, fopen(StrEq(path.c_str()), _)).Times(1).WillRepeatedly(Return(MOCK_FILE_DESCRIPTOR));;
```
Comment (mgonchar) [inline tests/mfx_dispatch/linux/mfx_dispatch_test_cases_plugins.cpp]: delete one ```;```

File: tests/mfx_dispatch/linux/mfx_dispatch_test_cases_plugins.cpp [hunk 4]
```diff
@@ -80,6 +95,140 @@ TEST_P(DispatcherPluginsTestParametrized, ShouldLoadFromGoodPluginsCfgCorrectly)
...
+TEST_F(DispatcherPluginsTest, ShouldFailIfPluginHasNoSymbols)
+{
+    MockCallObj& mock = *g_call_obj_ptr;
+    SetEmulatedPluginsCfgContent(mock, TEST_PLUGIN_COUNT);
+
+    const PluginParamExtended& plugin = m_plugin_vec[0];
+
+    if (!successfully_parsed_at_least_once)
+    {
+        std::string path = std::string(MFX_PLUGINS_CONF_DIR) + "/plugins.cfg";
```
Comment (mgonchar) [inline tests/mfx_dispatch/linux/mfx_dispatch_test_cases_plugins.cpp]: same

File: tests/mfx_dispatch/linux/mfx_dispatch_test_cases_plugins.cpp (hunk 3 above)
Comment (mgonchar) [inline tests/mfx_dispatch/linux/mfx_dispatch_test_cases_plugins.cpp]: looks like, you don't need ```std::string``` at all, ```MFX_PLUGINS_CONF_DIR "/plugins.cfg"``` is already a C-string which you can pass directly to ```fopen```

File: tests/mfx_dispatch/linux/mfx_dispatch_test_cases_plugins.cpp [hunk 5]
```diff
@@ -80,6 +95,140 @@ TEST_P(DispatcherPluginsTestParametrized, ShouldLoadFromGoodPluginsCfgCorrectly)
...
+TEST_F(DispatcherPluginsTest, ShouldFailIfPluginIsNotCreated)
+{
+    MockCallObj& mock = *g_call_obj_ptr;
+    SetEmulatedPluginsCfgContent(mock, TEST_PLUGIN_COUNT);
+
+    const PluginParamExtended& plugin = m_plugin_vec[0];
+
+    if (!successfully_parsed_at_least_once)
+    {
+        std::string path = std::string(MFX_PLUGINS_CONF_DIR) + "/plugins.cfg";
```
Comment (mgonchar) [inline tests/mfx_dispatch/linux/mfx_dispatch_test_cases_plugins.cpp]: same

File: tests/mfx_dispatch/linux/mfx_dispatch_test_cases_plugins.cpp [hunk 6]
```diff
@@ -33,14 +33,30 @@ TEST_F(DispatcherPluginsTest, ShouldSearchForPluginsCfgByCorrectPath)
 {
     MockCallObj& mock = *g_call_obj_ptr;
 
-    std::string path = std::string(MFX_PLUGINS_CONF_DIR) + "/plugins.cfg";
+    std::string path(MFX_PLUGINS_CONF_DIR "/plugins.cfg");
```
[the remark this answers is not in the document]
Comment (semelianova) [reply tests/mfx_dispatch/linux/mfx_dispatch_test_cases_plugins.cpp]: I agree, It looks better

**Pull request merged**